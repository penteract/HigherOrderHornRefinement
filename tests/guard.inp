environment
f : int -> bool ;

program
∃ p : (y:int->bool[y>0]). p x ⇒ f x ;

goal
E n:int. n<=1 ^ f n
