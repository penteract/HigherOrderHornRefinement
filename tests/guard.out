false \/ x_5 > 0 => x_2 x_5

goal:
E x_6:Int.x_6 <= 1 ^ x_2 x_6

